engine/src/main/java/io/zeebe/engine/processor/AsyncSnapshotDirector.java

```diff
@@ -78,17 +78,75 @@ protected void onActorStarting() {
     commitCondition = actor.onCondition(getConditionNameForPosition(), this::onCommitCheck);
     logStream.registerOnCommitPositionUpdatedCondition(commitCondition);
 
-    lastValidSnapshotPosition = snapshotController.getLastValidSnapshotPosition();
-    LOG.debug(
-        "The position of the last valid snapshot is '{}'. Taking snapshots beyond this position.",
-        lastValidSnapshotPosition);
+    lastValidSnapshotPosition = -1;
+    actor.runOnCompletionBlockingCurrentPhase(
+        streamProcessor.getLastProcessedPositionAsync(),
+        (position, error) -> {
+          if (error == null) {
+            lastValidSnapshotPosition = position;
+            LOG.debug(
+                "The position of the last valid snapshot is '{}'. Taking snapshots beyond this position.",
+                lastValidSnapshotPosition);
+          } else {
+            LOG.error(ERROR_MSG_ON_RESOLVE_PROCESSED_POS, error);
+          }
+        });
   }
 
   @Override
   protected void onActorCloseRequested() {
     logStream.removeOnCommitPositionUpdatedCondition(commitCondition);
   }
 
+  @Override
+  public ActorFuture<Void> closeAsync() {
```

Comment:
We have to check our format settings again. In the latest reviews I often saw methods rearrangements. It is is really hard to follow.

Reply:
Yes, I reverted those, not sure why it rearranged them :s